### PR TITLE
trace-agent: tag receiver and processing telemetry with service

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

AIT-9470

Tag trace-agent's receiver and processing telemetry metrics with `service` tag.

### Motivation

Ease debugging & supportability.

### Additional Notes

Writers' telemetry metrics are not covered since the written payloads can contain multiple services in the same payload.

### Describe how to test/QA your changes

E2e tests.

### CODE REVIEW

File: pkg/trace/api/api.go [hunk 1]
```diff
@@ -456,6 +456,14 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	firstService := func(cs *pb.ClientStatsPayload) string {
```
Comment (ajgajg1134) [inline pkg/trace/api/api.go:459]: curious why this is a small helper function here?

**Review by ajgajg1134 — APPROVED** [review]
Overall this looks good! Also nice to see how concise the new e2e tests are for this change!!
Just one comment curious why you went with an inline helper function that only is used once - not sure if I missed a reason why it was required or not. In any case it doesn't need to change I'm just asking out of curiosity

File: pkg/trace/api/api.go (hunk 1 above)
Comment (ahmed-mez) [reply pkg/trace/api/api.go:459]: Good question! At first, I wrote these 2 helpers as separate functions - unfortunately there is no straightforward way to write a generic one that covers both stats and traces. `handleStats` is relatively simple but `handleTraces` is quite long and does too many things, that's why I wanted the logic to be scoped in a helper even if it's used once.

Then, I thought it'd be a good idea to "protect" the package from relying on these helpers to get the service name for critical features in the future, that's why I encapsulated them as nested functions.

**Pull request merged**